tests/network/port_forward.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libvmi"
+)
+
+var _ = SIGDescribe("[Serial] Port-forward", func() {
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+
+		tests.BeforeTestCleanup()
+	})
+
+	Context("VMI With masquerade binding", func() {
+		const (
+			localPort       = 1500
+			declaredPort    = 1501
+			nonDeclaredPort = 1502
+		)
+		var portForwardCmd *exec.Cmd
+
+		When("performing port-forward from a local port to a VMI's pod declared port", func() {
+			var (
+				ports = []v1.Port{
+					{Port: declaredPort},
+				}
+				vmiWithDeclaredPort *v1.VirtualMachineInstance
+			)
+
+			BeforeEach(func() {
+				vmiWithDeclaredPort = createCirrosVMIWithPortsAndBlockUntilReady(virtClient, ports)
+				tests.StartHTTPServer(vmiWithDeclaredPort, declaredPort)
+
+				vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmiWithDeclaredPort, tests.NamespaceTestDefault)
+				portForwardCmd, err = portForwardCommand(vmiPod, localPort, declaredPort)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(portForwardCmd.Start()).To(Succeed())
+			})
+
+			It("should reach the vmi", func() {
+				defer func() {
+					Expect(killPortForwardCommand(portForwardCmd)).To(Succeed())
+				}()
+
+				By(fmt.Sprintf("checking that service running on port %d can be reached", declaredPort))
+				Eventually(func() error {
+					err = exec.Command("curl", fmt.Sprintf("127.0.0.1:%d", localPort)).Run()
+					return err
+				}, 40*time.Second, 5*time.Second).ShouldNot(HaveOccurred())
+			})
+		})
+
+		When("performing port-forward from a local port to a VMI's pod non-declared port", func() {
+			var vmiWithoutDeclaredPort *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmiWithoutDeclaredPort = createCirrosVMIWithPortsAndBlockUntilReady(virtClient, []v1.Port{})
```

Comment:
This BeforeEach and the one on line 68 is practically the same.

What do you think about moving the body of this `BeforeEach` it to an upper level `JustBeforeEach`?
Then you could have this `BeforeEach` (and the one on line 68)  to only set `ports` variable accordingly - either with an explicit port or without any ports.

Reply:
ahh that's a lot nicer. Done.
What do think about it now?